test/new-e2e/tests/agent-platform/common/agent_behaviour.go

```diff
@@ -14,13 +14,67 @@ import (
 	"time"
 
 	e2eClient "github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
+	"github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-platform/common/bound-port"
+	"github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-platform/common/process"
 	"github.com/stretchr/testify/require"
 )
 
+// RunningAgentProcesses returns the list of running agent processes
+func RunningAgentProcesses(client *TestClient) ([]string, error) {
+	agentProcesses := client.Helper.AgentProcesses()
+	runningAgentProcesses := []string{}
+	for _, process := range agentProcesses {
+		if AgentProcessIsRunning(client, process) {
+			runningAgentProcesses = append(runningAgentProcesses, process)
+		}
+	}
+	return runningAgentProcesses, nil
+}
+
+// AgentProcessIsRunning returns true if the agent process is running
+func AgentProcessIsRunning(client *TestClient, processName string) bool {
+	running, err := process.IsProcessRunning(client.VMClient, processName)
+	return running && err == nil
+}
+
+// PortBoundByPID returns the info about the port bound by a given PID
+func PortBoundByPID(client *TestClient, port int, pid int) (boundport.BoundPort, error) {
+	ports, err := boundport.BoundPorts(client.VMClient)
+	if err != nil {
+		return nil, err
+	}
+	for _, boundPort := range ports {
+		if boundPort.PID() == pid && boundPort.LocalPort() == port {
+			return boundPort, nil
+		}
+	}
+	return nil, nil
+}
+
+// PortBoundByService returns info about the port bound by a given service
+func PortBoundByService(client *TestClient, port int, service string) (boundport.BoundPort, error) {
+	// TODO: might need to map service name to process name, this is working right now though
+	pids, err := process.FindPID(client.VMClient, service)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, pid := range pids {
+		boundPort, err := PortBoundByPID(client, port, pid)
+		if err != nil {
+			return nil, err
+		}
+		if boundPort != nil {
+			return boundPort, nil
+		}
+	}
+	return nil, nil
+}
```

Comment:
In `agent_behaviour.go` it would be nice to keep only "tests" for the behavior of the agent.  I think these util functions could be moved to `test_client.go` where functions like `SetPythonVersion` and other live

Reply:
moved